client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -95,6 +97,52 @@ tie.factory('PythonCodePreprocessorService', [
       return firstLine + '\n' + subsequentLines.join('\n');
     };
 
+    // Adds support for helper functions in student code by
+    // dynamically inserting the StudentCode class name into their code
+    // during the preprocessing phase, allowing it to run normally without
+    // any indication on the student side that this is happening.
+    var _addClassWrappingToStudentHelperFunctions = function(code){
+      while (true){
+        // Returns ['matching string', 'capture group']
+        var regexResult = PYTHON_FUNCTION_DEF_REGEX.exec(code);
+        if (!regexResult) {
+          return code;
+        }
+        // We should make sure that we don't append our class name to
+        // the student's function definition.
+        var functionNameStart = (code.indexOf(regexResult[0]) + 
+          regexResult[0].indexOf(regexResult[1]));
+        var functionName = regexResult[1];
+        lastFunctionNameLocation = 0;
+        while (true){
+          var codeToAdd = '';
```

Comment:
I define it there so that, if it gets skipped, codeToAdd's length is 0, meaning only 1 will get added to lastFunctionNameLocation. I'm happy to move it to functionPrefix, but I currently want to keep the logic of either adding 1 or functionPrefix.length + 1.